src/cmds/qsub.c

```diff
@@ -4087,13 +4088,16 @@ copy_env_value(char *dest, /* destination  */
 
 			case ESC_CHAR:			/* backslash in value, escape it */
 				*dest++ = *pv;
-				*dest++ = *pv;
+				if (*(pv+1) != ',') /* do not escape if ESC_CHAR already escapes */
+					*dest++ = *pv;
 				break;
 
 			case ',':
 				if (q_ch || quote_flg) {
 					*dest++ = ESC_CHAR;
 					*dest++ = *pv;
+				} else if (dest_full != dest && *(dest-1) == ESC_CHAR) { /* the comma is escaped, not finished yet */
+					*dest++ = *pv;
```

Comment:
I guess there's really nothing we can do here if a comma is the first character, so we fall through to the else. Works for me.